pkg/cmd/cli/cmd/process.go

```diff
@@ -171,73 +171,68 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
-		if !ok {
-			sourceName := filename
-			if len(templateName) > 0 {
-				sourceName = namespace + "/" + templateName
-			}
-			fmt.Fprintf(cmd.Out(), "unable to parse %q, not a valid Template but %s\n", sourceName, reflect.TypeOf(infos[i].Object))
-			continue
-		}
+	if len(infos) > 1 {
+		fmt.Fprintf(out, "%d input templates found, but only the first will be processed", len(infos))
+	}
 
-		// If 'parameters' flag is set it does not do processing but only print
-		// the template parameters to console for inspection.
-		// If multiple templates are passed, this will print combined output for all
-		// templates.
-		if kcmdutil.GetFlagBool(cmd, "parameters") {
-			if len(infos) > 1 {
-				fmt.Fprintf(out, "\n%s:\n", obj.Name)
-			}
-			if err := describe.PrintTemplateParameters(obj.Parameters, out); err != nil {
-				fmt.Fprintf(cmd.Out(), "error printing parameters for %q: %v\n", obj.Name, err)
-			}
-			continue
+	obj, ok := infos[0].Object.(*templateapi.Template)
+	if !ok {
+		sourceName := filename
+		if len(templateName) > 0 {
+			sourceName = namespace + "/" + templateName
 		}
+		return fmt.Errorf("unable to parse %q, not a valid Template but %s\n", sourceName, reflect.TypeOf(infos[0].Object))
+	}
 
-		if label := kcmdutil.GetFlagString(cmd, "labels"); len(label) > 0 {
-			lbl, err := kubectl.ParseLabels(label)
-			if err != nil {
-				fmt.Fprintf(cmd.Out(), "error parsing labels: %v\n", err)
-				continue
-			}
-			if obj.ObjectLabels == nil {
-				obj.ObjectLabels = make(map[string]string)
-			}
-			for key, value := range lbl {
-				obj.ObjectLabels[key] = value
-			}
-		}
+	// If 'parameters' flag is set it does not do processing but only print
+	// the template parameters to console for inspection.
+	if kcmdutil.GetFlagBool(cmd, "parameters") {
+		return describe.PrintTemplateParameters(obj.Parameters, out)
+	}
 
-		// Override the values for the current template parameters
-		// when user specify the --value
-		if cmd.Flag("value").Changed {
-			values := kcmdutil.GetFlagStringSlice(cmd, "value")
-			injectUserVars(values, cmd, obj)
+	if label := kcmdutil.GetFlagString(cmd, "labels"); len(label) > 0 {
+		lbl, err := kubectl.ParseLabels(label)
+		if err != nil {
+			return fmt.Errorf("error parsing labels: %v\n", err)
 		}
-		injectUserVars(valueArgs, cmd, obj)
+		if obj.ObjectLabels == nil {
+			obj.ObjectLabels = make(map[string]string)
+		}
+		for key, value := range lbl {
+			obj.ObjectLabels[key] = value
+		}
+	}
 
-		resultObj, err := client.TemplateConfigs(namespace).Create(obj)
-		if err != nil {
-			fmt.Fprintf(cmd.Out(), "error processing the template %q: %v\n", obj.Name, err)
-			continue
+	// Override the values for the current template parameters
+	// when user specify the --value
+	if cmd.Flag("value").Changed {
+		values := kcmdutil.GetFlagStringSlice(cmd, "value")
+		if err := injectUserVars(values, obj); err != nil {
+			return err
 		}
+	}
+	if err := injectUserVars(valueArgs, obj); err != nil {
```

Comment:
It doesn't have to be done here but we should be supporting key-value pairs
on the CLI for process like we do on env.

On Mon, Apr 18, 2016 at 8:36 AM, Steve Kuznetsov <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/process.go
> <https://github.com/openshift/origin/pull/8049#discussion_r60049723>:
>
> >  		}
> > +	}
> > +	if err := injectUserVars(valueArgs, obj); err != nil {
>
> Seems like oc process --value=foo=bar --value=foo=baz will go to baz as
> we just replace sequentially.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8049/files/26bb43368a6d80c201102b06de6e8bc93fbfc2ad#r60049723>
>